blobstore/blobhandler.go

```diff
@@ -305,3 +305,22 @@ func (bh *BlobHandler) PingWithAuth(c echo.Context) error {
 
 	return c.JSON(http.StatusOK, bucketHealth)
 }
+
+func (bh *BlobHandler) HandleCheckS3UserPermission(c echo.Context) error {
+	if bh.Config.AuthLevel == 0 {
```

Comment:
There is some more logic needed here -- if there is not fine grained access control, that does not mean every user can upload, it just means that we are not limited the upload by bucket. A user could be s3_reader and still not be able to upload.

Reply:
The user will not be able to call this endpoint unless they have s3_limited_writer or s3_writer in their roles (it's a secondary check in the front end after checking their writer roles). So I can make those roles and the admin the only people that can call this endpoint!